ext/nnstreamer/tensor_filter/tensor_filter_tvm.cc

```diff
@@ -0,0 +1,413 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * NNStreamer tensor_filter, sub-plugin for TVM
+ * Copyright (C) 2021 Junhwan Kim <jejudo.kim@samsung.com>
+ */
+/**
+ * @file    unittest_filter_tvm.cc
+ * @date    16 Apr 2021
+ * @brief   NNStreamer tensor-filter sub-plugin for Apache TVM
+ * @author  Junhwan Kim <jejudo.kim@samsung.com>
+ * @see     http://github.com/nnstreamer/nnstreamer
+ * @bug     No known bugs
+ * 
+ * This is the per-NN-framework plugin (TVM) for tensor_filter.
+ * 
+ * @note    Only supports tvm.contrib.graph_executor model
+ */
+
+#include <glib.h>
+#include <nnstreamer_cppplugin_api_filter.hh>
+#include <nnstreamer_log.h>
+#include <tensor_common.h>
+
+#include <dlpack/dlpack.h>
+#include <tvm/runtime/module.h>
+#include <tvm/runtime/packed_func.h>
+#include <tvm/runtime/registry.h>
+
+
+namespace nnstreamer
+{
+namespace tensorfilter_tvm
+{
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+void init_filter_tvm (void) __attribute__ ((constructor));
+void fini_filter_tvm (void) __attribute__ ((destructor));
+#ifdef __cplusplus
+}
+#endif /* __cplusplus */
+
+
+/**
+ * @brief Tensor data information to execute the model
+ */
+typedef struct {
+  std::vector<int64_t> shape;
+  DLDataType dtype;
+} tvm_data_info;
+
+/**
+ * @brief Class for TVM subplugin.
+ */
+class tvm_subplugin final : public tensor_filter_subplugin
+{
+  private:
+  bool empty_model;
+  char *model_path;
+  GstTensorsInfo inputInfo;
+  GstTensorsInfo outputInfo;
+
+  DLDevice device;
+  tvm::runtime::Module mod_factory;
+  tvm::runtime::Module gmod;
+  std::vector<tvm_data_info> input_info_list;
+  std::vector<tvm_data_info> output_info_list;
+
+  static const char *name;
+  static const accl_hw hw_list[];
+  static tvm_subplugin *registeredRepresentation;
+
+  bool parse_custom_prop (const char *custom_prop);
+  bool convert_nns_type (tensor_type nns_type, DLDataType &tvm_type);
```

Comment:
how about passing nns_type by const reference?

```suggestion
  bool convert_nns_type (const tensor_type& nns_type, DLDataType &tvm_type);
```